libs/chain-signatures/contract/src/primitives/code_hash.rs

```diff
@@ -0,0 +1,47 @@
+use near_sdk::{log, near};
+use std::collections::BTreeMap;
+
+use super::key_state::AuthenticatedParticipantId;
+
+/// Hash of a Docker image running in the TEE environment.
+#[near(serializers=[borsh, json])]
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone)]
+pub struct CodeHash([u8; 32]);
+
+/// Tracks votes to add whitelisted TEE code hashes. Each participant can at any given time vote for
+/// a code hash to add.
+#[near(serializers=[borsh, json])]
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+pub struct CodeHashesVotes {
+    pub proposal_by_account: BTreeMap<AuthenticatedParticipantId, CodeHash>,
```

Comment:
what's the reason for doing it this way vs. `HashMap<CodeHash, HashSet<AuthenticatedParticipantId>>`? I guess there is not much real difference since we are talking about very small number of participants here but just curious

Reply:
Funny enough, I asked myself that too!

After digging into the code, I saw there are already a few places where voting is done this way. Probably means we could abstract it out and reuse it across different parts of the code.

https://github.com/Near-One/mpc/blob/a35bc500a2e29592c9aef818b5a31143e5f58715/libs/chain-signatures/contract/src/primitives/domain.rs#L167-L173

https://github.com/Near-One/mpc/blob/a35bc500a2e29592c9aef818b5a31143e5f58715/libs/chain-signatures/contract/src/v0_state/mod.rs#L25-L30

Probably @kevindeforth or @robin-near can help with a bit more context on this.